config/routes.rb

```diff
@@ -62,6 +62,8 @@
 
   get 'geographical_areas/:id', to: 'geographical_areas#show', as: :geographical_area
 
+  get '/measure_type/:measure_type_id/:preference_code_id', to: 'measure_type#show', as: :measure_type
```

Comment:
This feels weird to me. So the thing that uniquely identifies the measure type is the measure type id. Everything else should be a query param IMO

Reply:
I would strongly disagree on this point - the page requires a dual primary key - the combination of :measure_type_id and :perference_code_id, it cannot be rendered with either just one or just the other. Making either part optional (which query params should be) is counter intuitive and results in an unrenderable page. I also think having half the primary key in one format (url component) and half in another (query param) would be wierd.

An alternative to using a slash for the separator on primary key components would be to use a hypen - thats less common but at least keeps both primary keys in the same domain

Reply:
The alternative maybe to use a different base URL or to add a preference code segment

`/measure_type/:measure_type_id/preference_code/:preference_code_id`

Reply:
Your last comment was exactly what I was about to suggest, yeah. Let's go with that

Reply:
```
/measure_types/:measure_type_id/preference_codes/:id
```